libraries/resource_provider.rb

```diff
@@ -183,6 +183,50 @@ def create(method = :create)
       end
     end
 
+    # Adds Oneview Scope to the Oneview resource if scope is not already added
```

Comment:
It looks like these 3 methods are adding quite a few lines of code and are really similar.
Couldn't this be one generic method that receives the function you want to call?

Reply:
Yes, I'm thinking about this now.